### PR TITLE
Fix issue #57 Update frontend_user tests 

### PR DESCRIPTION
**Pull request by pritalpatel**
Pull Request for Issue #57 .

### CODE REVIEW

File: tests/_support/AcceptanceTester.php [hunk 1]
```diff
@@ -201,7 +204,7 @@ public function doAdministratorLogout()
 		$I->waitForElement(['xpath' => "//li[@class='dropdown open']/ul[@class='dropdown-menu']//a[text() = 'Logout']"], TIMEOUT);
 		$I->click(['xpath' => "//li[@class='dropdown open']/ul[@class='dropdown-menu']//a[text() = 'Logout']"]);
 		$I->waitForElement(['id' => 'mod-login-username'], TIMEOUT);
-		$I->waitForText('Log in', TIMEOUT, ['xpath' => "//fieldset[@class='loginform']//button"]);
+        $I->waitForText('Log in', TIMEOUT, ['class' => "btn-primary"]);
```
Comment (yvesh) [inline tests/_support/AcceptanceTester.php]: We have an indention issue here (and an empty new line after)

Comment (pritalpatel) [inline tests/_support/AcceptanceTester.php]: I will update the whole branch and will correct all the code style correction.

**Comment by pritalpatel** [comment]
I finished applying coding standards. Please review.

**Comment by pritalpatel** [comment]
@puneet0191  ready to merge.

**Comment by puneet0191** [comment]
Let @yvesh  review the new tests, I don't know what all scenarios you were supposed to cover.

File: RoboFile.php [hunk 2]
```diff
@@ -425,7 +463,8 @@ public function runTest($pathToTestFile = null, $suite = 'acceptance')
 			foreach ($class_methods as $method_name)
 			{
 				$reflect = new ReflectionMethod($className[0], $method_name);
-				if(!$reflect->isConstructor())
+
+				if (!$reflect->isConstructor())
```
Comment (yvesh) [inline RoboFile.php]: You can combine this too ifs (no need for nesting)

E.g. 

``` php
                if (!$reflect->isConstructor())
                {
                    if ($reflect->isPublic())
                    {
```

becomes

``` php
                if (!$reflect->isConstructor() && $reflect->isPublic())
                {
```

File: tests/codeception/_support/Step/Acceptance/Site/UsersFrontend.php [hunk 3]
```diff
@@ -200,6 +200,7 @@ public function iShouldSeeMessage($message)
 	public function userIsCreated()
 	{
 		$I = $this;
+		$I->comment('User is created');
```
Comment (yvesh) [inline tests/codeception/_support/Step/Acceptance/Site/UsersFrontend.php:212]: We still need the check here, don't we? :)

**Comment by pritalpatel** [comment]
@yvesh I have sent fix for verifying user creation.

File: tests/codeception/_support/Step/Acceptance/Site/UsersFrontend.php [hunk 4]
```diff
@@ -490,7 +500,7 @@ public function iShouldSeeTheName($name)
 	 *
 	 * @param   string  $name  The name of the user
 	 *
-	 * @Given Needs to user :arg1 logged in at least once
+	 * @When    I Needs to user :arg1 logged in at least once
```
Comment (nibra) [inline tests/codeception/_support/Step/Acceptance/Site/UsersFrontend.php]: This is no valid English...

```
@Given User :arg1 did login at least once
```

File: tests/codeception/acceptance/users_frontend.feature [hunk 5]
```diff
@@ -45,6 +46,7 @@ Feature: Users Frontend
     Then I should see the name "User Three"
 
   Scenario: Test last login date
-    Given Needs to user "User Two" logged in at least once
-    When I login as a super admin from backend
+    Given I am on the User Manager page
+    When I Needs to user "User Two" logged in at least once
+    And I login as a super admin from backend
```
Comment (nibra) [inline tests/codeception/acceptance/users_frontend.feature]: `And I login as a super admin from backend` is already covered by `Given I am on the User Manager page`, isn't it?

``` Gherkin
Given  I am on the User Manager page
And    User :arg1 did login at least once
```

File: tests/codeception/acceptance/users_frontend.feature [hunk 6]
```diff
@@ -45,6 +46,7 @@ Feature: Users Frontend
...
     Then I should see last login date for "User Three"
```
Comment (nibra) [inline tests/codeception/acceptance/users_frontend.feature:52]: Shouldn't it be "User Two"?

**Comment by pritalpatel** [comment]
@nibra I have fixed your comments.

**Pull request merged**